### PR TITLE
## Bookstore: connect to API

### PR DESCRIPTION
**Pull request by abiodunraheem**
## Bookstore: connect to API

- I used the concept of state management.
- Used store, actions, and reducers in React.
- Connected React and Redux.
- I connected an existing API via Redux and use the data to display books in the UI.

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+      </div>
+
+      <div className="book-ch">
+        <span className="current-chapt">CURRENT CHAPTER</span>
+        <span className="chapt">Chapter 17</span>
+        <button type="button" className="update-btn">UPDATE PROGRESS</button>
+      </div>
+    </div>
   );
 };
```
Comment (Ismailco) [inline src/components/Books.js]: - [ ] Great job creating the app however there are some errors on the console, please consider fixing them first. so your app has no errors.
   - In the prop validation, you are checking for an object, but in the props, you're passing an array,
   - also there are duplicated keys on your array, so you can use Math.random() or use the UUID to generate a random key

Screenshot_20220617_004832

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @abiodunraheem,

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+import categories from './categories/categories';
 
-const reducer = combineReducers({
-  bookReducer,
-  checkStatusReducer,
+const store = configureStore({
+  reducer: {
+    books: books.reducer,
+    categories: categories.reducer,
+  },
```
Comment (roshan-bajgain) [inline src/redux/configureStore.js:9]: - [ ] [Optional] Kindly add redux-thunk to the project as a package and use it as a middleware. Eg:
```
  import { createStore, applyMiddleware } from 'redux';
    import thunk from 'redux-thunk';
    import rootReducer from './reducers/index';

    const store = createStore(
        rootReducer,
        applyMiddleware(thunk)
    );
```

**Review by roshan-bajgain — APPROVED** [review]
Hi @abiodunraheem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**